### PR TITLE
Replace deprecated RSA function call

### PR DESCRIPTION
**Pull request by sachaaaaa**

### CODE REVIEW

File: httpserver/server_certificates.h [hunk 1]
```diff
@@ -86,9 +87,18 @@ int mkcert(X509** x509p, EVP_PKEY** pkeyp, int bits, int serial, int days) {
     } else
         x = *x509p;
 
-    rsa = RSA_generate_key(bits, RSA_F4, NULL, NULL);
+    bne = BN_new();
```
Comment (msgmaxim) [inline httpserver/server_certificates.h]: Does this need `BN_free`?

File: httpserver/server_certificates.h [hunk 2]
```diff
@@ -86,9 +87,18 @@ int mkcert(X509** x509p, EVP_PKEY** pkeyp, int bits, int serial, int days) {
...
+    rsa = RSA_new();
```
Comment (msgmaxim) [inline httpserver/server_certificates.h]: Similarly, does this need `RSA_free`?

File: httpserver/server_certificates.h (hunk 1 above)
Comment (sachaaaaa) [reply httpserver/server_certificates.h]: Yes! Thanks...

File: httpserver/server_certificates.h (hunk 2 above)
Comment (sachaaaaa) [reply httpserver/server_certificates.h]: Needed to do some digging, but basically no. The rsa key is "attached" to the EVP_PKEY with `EVP_PKEY_assign_RSA`, so the EVP_PKEY takes care of freeing the rsa key...
https://www.openssl.org/docs/man1.0.2/man3/EVP_PKEY_assign_RSA.html

**Pull request merged**